process/smartContract/hooks/blockChainHook.go

```diff
@@ -404,6 +422,26 @@ func (bh *BlockChainHookImpl) NewAddress(creatorAddress []byte, creatorNonce uin
 func (bh *BlockChainHookImpl) ProcessBuiltInFunction(input *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
 	defer stopMeasure(startMeasure("ProcessBuiltInFunction"))
 
+	if bh.enableEpochsHandler.IsMaxBlockchainHookCountersFlagEnabled() {
+		gasSchedule := bh.gasSchedule.LatestGasSchedule()
+
+		maxBuiltInCallsPerTx := gasSchedule["MaxPerTransaction"]["MaxBuiltInCallsPerTx"]
+		bh.crtNumberOfBuiltInFunctionCalls++
+		if bh.crtNumberOfBuiltInFunctionCalls > maxBuiltInCallsPerTx {
+			return nil, process.OutOfAPICalls
+		}
+
+		maxNumberOfTransfersPerTx := gasSchedule["MaxPerTransaction"]["MaxNumberOfTransfersPerTx"]
+		parsedTransfer, err := bh.esdtTransferParser.ParseESDTTransfers(input.CallerAddr, input.RecipientAddr, input.Function, input.Arguments)
+		if err != nil {
+			return nil, err
```

Comment:
you should not error here - it is the job of the built in function to error on that case. in case of error you just leave crtNumberOfTransfers as it is.

Reply:
done